nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js

```diff
@@ -774,13 +775,37 @@ export function RanEdit(props: Props) {
   const [connectedEnodebs, setConnectedEnodebs] = useState<enodeb_serials>(
     props.gateway.connected_enodeb_serials,
   );
+  const [unregisteredEnodebs, setUnregisteredEnodebs] = useState({});
   const handleRanChange = (key: string, val) => {
     setRanConfig({...ranConfig, [key]: val});
   };
   const handleDnsChange = (key: string, val) => {
     setDnsConfig({...dnsConfig, [key]: val});
   };
-
+  useEffect(() => {
+    const isEnodebUnregistered = (enb: enodeb, currentGateway: lte_gateway) => {
+      const gatewaysList = Object.keys(ctx.state);
+      for (const gatewayId of gatewaysList) {
+        if (
+          ctx.state[gatewayId].connected_enodeb_serials?.includes(enb.serial) &&
+          ctx.state[gatewayId].id != currentGateway.id
+        ) {
+          return false;
+        }
+      }
+      return true;
+    };
+    const newUnregisteredEnodebs = {};
+    if (enbsCtx?.state?.enbInfo) {
+      Object.keys(enbsCtx.state.enbInfo).map(enbSerial => {
+        newUnregisteredEnodebs[enbSerial] = isEnodebUnregistered(
```

Comment:
You can make the map method return the list of unregisteredEnbInfo and directly use it as 
`Object.keys(unregisteredEnbInfo).map(enbSerial =`

Reply:
I just did that. I mapped through unregisteredEnbsInfo and show only unregistered enodebs and removed the if else logic.